### PR TITLE
moved character details into components and refactored imports 

### PR DESCRIPTION
**Pull request by 100xlongx**

### CODE REVIEW

File: src/app/components/character-details/character-details.component.spec.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-import { CharacterService } from '../services/character.service';
+import { CharacterServiceAbstract, MockCharacterService } from '../../services/character-service-abstract';
+import { CharacterService } from '../../services/character.service';
 
 import { CharacterDetailsComponent } from './character-details.component';
 
 import { of } from "rxjs";
 
-import { routes } from "../app-routing.module";
+import { routes } from "../../app-routing.module"
```
Comment (mosdevly) [inline src/app/components/character-details/character-details.component.spec.ts]: Missing semicolon.

File: src/app/components/character-details/character-details.component.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Component, OnInit } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import Character from '../models/Character';
-import { CharacterService } from '../services/character.service';
+import Character from '../../models/Character';
+import { CharacterService } from '../../services/character.service'
```
Comment (mosdevly) [inline src/app/components/character-details/character-details.component.ts]: Missing semicolon.

**Review by LWooler — APPROVED** [review]
excellent changes

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 import { RouterModule, Routes } from '@angular/router';
-import { CharacterDetailsComponent } from './character-details/character-details.component';
+import { CharacterDetailsComponent } from './components/character-details/character-details.component';
```
Comment (mosdevly) [inline src/app/app-routing.module.ts:4]: yes! :)

**Review by mosdevly — CHANGES_REQUESTED** [review]

**Review by mosdevly — APPROVED** [review]

**Pull request merged**